### PR TITLE
[CELEBORN-1051] Add `isDynamic` property for CelebornConf

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Since we support ConfigService, many configuration can be dynamic, add `isDynamic` property for CelebornConf in this pr.

### Why are the changes needed?
Make configuration doc more cleear

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT

### CODE REVIEW

File: common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala [hunk 1]
```diff
@@ -139,8 +140,8 @@ class ConfigurationSuite extends AnyFunSuite {
   }
 
   def appendConfigurationTableHeader(output: ArrayBuffer[String]): Unit = {
-    output += "| Key | Default | Description | Since | Deprecated |"
-    output += "| --- | ------- | ----------- | ----- | ---------- |"
+    output += "| Key | Default | isDynamic | Description | Since | Deprecated |"
```
Comment (SteNicholas) [inline common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala:143]: Why does the document show whether the config is dynamic? IMO, all configs could be dynamic at default.

**Comment by SteNicholas** [comment]
@AngersZhuuuu, could you confirm which configs support dynamic configuration in this pull request? IMO, most configs of `CelebornConf` could support dynamic configuration. Otherwise, whether the config support dynamic configuration is case by case to define.

**Review by SteNicholas — APPROVED** [review]
@AngersZhuuuu, LGTM. BTW, could you add some description in config document to describe this property?

**Review by RexXiong — APPROVED** [review]
LGTM, thanks!

File: docs/migration.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ license: |
 
 - Since 0.5.0, Celeborn worker metrics `CreditStreamCount` is renamed as `ActiveCreditStreamCount`.
 
+- Since 0.5.0, Celeborn configurations support new tag `isDynamic` to represent whether the configuration is dynamic config.
```
Comment (pan3793) [inline docs/migration.md:30]: IMO it's not a breaking change, we don't need this.

**Pull request closed without merging**